api/internal/generators/utils.go

```diff
@@ -36,25 +36,37 @@ kind: %s
 	return rn, nil
 }
 
+type orderedMap struct {
+	keys   []string
+	values map[string]string
+}
+
 func makeValidatedDataMap(
-	ldr ifc.KvLoader, name string, sources types.KvPairSources) (map[string]string, error) {
+	ldr ifc.KvLoader, name string, sources types.KvPairSources,
+) (orderedMap, error) {
 	pairs, err := ldr.Load(sources)
 	if err != nil {
-		return nil, errors.WrapPrefix(err, "loading KV pairs", 0)
+		return orderedMap{}, errors.WrapPrefix(err, "loading KV pairs", 0)
 	}
-	knownKeys := make(map[string]string)
+
+	seen := make(map[string]struct{})
```

Comment:
Why do we need the `seen` map? The `values` map will always contain the same keys, so why not use that for checking duplicates?